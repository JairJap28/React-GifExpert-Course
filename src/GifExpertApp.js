import React, { useState } from 'react';

import AddCategory from './components/AddCategory';
import EmptySearch from './components/GifEmptySearch';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {
    const [categories, setCategories] = useState([]);

    const handleAdd = (value) => {
        setCategories([value, ...categories]);
    }

    return (
        <>
            <h2 className="title">GifExpertApp</h2>
            <AddCategory setCategories={handleAdd}/>
            <hr/>
            { 
                categories && categories.length > 0 ? 
                <ol>{ categories.map(item => <GifGrid key={item} category={item}/>) } </ol>
                :
                <EmptySearch />
            }
        </>
    );
}

export default GifExpertApp;