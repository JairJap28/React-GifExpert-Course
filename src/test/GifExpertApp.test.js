import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Test on GifExpertApp component', () => {
    test('Should match with the snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Should show the empty component when it does not have categories', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper.find('EmptySearch').exists()).toBe(true);
    });

    test('Should show a list of categories', () => {
        const categories = ['The Flash', 'Superman'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        expect(wrapper.find('EmptySearch').exists()).toBe(false);
    });
});