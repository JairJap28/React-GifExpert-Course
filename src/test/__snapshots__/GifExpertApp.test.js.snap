// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifExpertApp component Should match with the snapshot 1`] = `
<Fragment>
  <h2
    className="title"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <EmptySearch />
</Fragment>
`;

exports[`Test on GifExpertApp component Should show a list of categories 1`] = `
<Fragment>
  <h2
    className="title"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="The Flash"
      key="The Flash"
    />
    <GifGrid
      category="Superman"
      key="Superman"
    />
     
  </ol>
</Fragment>
`;
